### PR TITLE
luci-app-advanced-reboot: convert to client side

### PR DESCRIPTION
**Pull request by Ansuel**
Rework the advanced-reboot app to client side implementation.
Change devices file to json type.

This still needs to be tested since I worked on this remotely. (The reboot to a new partition and the power off feature is not tested.)

Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>

### CODE REVIEW

**Comment by pkgadd** [comment]
This seems to fails with
```
RPCError

RPC call to luci.advanced_reboot/obtain_device_info failed with error -32000: Object not found
  at handleCallReply (https://nbg6817.xxx.example.com/luci-static/resources/rpc.js?v=git-20.278.53893-c2b22af:11:3)
```
patch also complains about trailing whitespace:
```
Warning: trailing whitespace in lines 36,89,93,100,113 of feeds/luci/applications/luci-app-advanced-reboot/htdocs/luci-static/resources/view/system/advanced_reboot.js
Warning: trailing whitespace in lines 19,32,34,66,75,92,139,146,150,162,187 of feeds/luci/applications/luci-app-advanced-reboot/root/usr/libexec/rpcd/luci.advanced_reboot
```

**Comment by Ansuel** [comment]
You restarted rpcd?

Il Lun 5 Ott 2020, 04:46 pkgadd <notifications@github.com> ha scritto:

**Comment by Ansuel** [comment]
(about whitespace, this is still WIP and I still need to clean this up)

**Comment by pkgadd** [comment]
It was a complete, fresh, build from scratch with the patch applied - exec permissions might have been missed though (courtesy of patch && quilt). I'll do a new build making sure of that in the evening, when I get home.

**Comment by Ansuel** [comment]
Could also be my fault not setting it in git... now should be good on my part

**Comment by stangri** [comment]
@Ansuel amazing work! I'll try to build an ipk, install it on my router and experiment some time this week.

**Comment by 1Jeff1** [comment]
Tested on the wrt32x ,when it reboot to the alternate partition it return to the same partition, the `Proceed` button does not respond on the power off feature.

**Comment by Ansuel** [comment]
@1Jeff1 thx for the feedback... all the problems should be fixed... pls retest when you can

**Comment by 1Jeff1** [comment]
Tested again, reboot to alternate partition is working properly now, power off feature works but is missing the message `The system is shutting down now. DO NOT POWER OFF THE DEVICE!`

**Comment by Ansuel** [comment]
Perfect... I need to fix this and check that translations works and this is done!

**Comment by Ansuel** [comment]
@jow- can you review this?

File: applications/luci-app-advanced-reboot/root/etc/uci-defaults/40_luci-advanced-reboot [hunk 1]
```diff
@@ -1,5 +1,6 @@
 #!/bin/sh
 
+/etc/init.d/rpcd restart
```
Comment (jow-) [inline applications/luci-app-advanced-reboot/root/etc/uci-defaults/40_luci-advanced-reboot:3]: A reload should be enough.

**Comment by hnyman** [comment]
I tested 6d1ce11 with my WRT3200ACM and both a plain reboot and a boot to alternative worked ok.  ;-)
(master r14641-582ba9ddc7)

**Comment by pkgadd** [comment]
The current version (6d1ce1160c45c7e68d312e4e0c48cfe6caadf4e7) seems to work fine on my nbg6817 as well.

Just one tiny (cosmetic) difference, the partition number is displayed as hexadecimal value in the current lua version:
nbg6817 - Advanced Reboot - LuCI-lua

while it's interpreted as decimal in the new javascript based implementation:
nbg6817 - Advanced Reboot - LuCI-javascript

imho hexadecimal makes more sense here, but that's mostly cosmetic.

```
Tested-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
```

**Comment by Ansuel** [comment]
Easy fix...

**Comment by Ansuel** [comment]
Done ;)

**Comment by pkgadd** [comment]
Tested (21a13cb76f0bb6be229cc3de30b4d25ededfe964) and confirmed to be working, thanks a lot.

**Comment by stangri** [comment]
@Ansuel outstanding job, feel free to merge any time.

**Comment by hnyman** [comment]
@Ansuel 
is this now ready to be merged now?
(or still some tweaks to come?)

**Comment by Ansuel** [comment]
Yes it's ready. They tester both linksys and Netgear variant.

Il Mer 7 Ott 2020, 19:16 Hannu Nyman <notifications@github.com> ha scritto:

**Comment by brvphoenix** [comment]
@Ansuel 
Does this package still depend on luci-compat?

**Pull request merged**